extensions/resource-deployment/src/test/services/platformService.test.ts

```diff
@@ -0,0 +1,343 @@
+					stub.getCall(0).args[0].should.equal(filePath);
+					if (ignoreError) {
+						should(error).be.undefined();
+					} else {
+						should(error).not.be.undefined();
+					}
+				});
+			});
+		});
+		it('openFile', () => {
+			const stub = sinon.stub(vscode.commands, 'executeCommand').resolves(); //resolves with a known string
+			platformService.openFile(filePath);
+			stub.callCount.should.equal(1);
+			stub.getCall(0).args[0].should.equal('vscode.open');
+			stub.getCall(0).args[1].path.should.equal(filePath);
+		});
+		it('readTextFile', async () => {
+			sinon.stub(fs.promises, 'readFile').resolves(contents);
+			const result = await platformService.readTextFile(filePath);
+			result.should.equal(contents);
+		});
+		it('saveTextFile', async () => {
+			const stub = sinon.stub(fs.promises, 'writeFile').resolves(); //resolves with a known string
+			await platformService.saveTextFile(contents, filePath);
+			stub.callCount.should.equal(1);
+			stub.getCall(0).args[0].should.equal(filePath);
+			stub.getCall(0).args[1].should.equal(contents);
+		});
+		it('copyFile', async () => {
+			const target = __dirname; //arbitrary path
+			const stub = sinon.stub(fs.promises, 'copyFile').resolves();
+			await platformService.copyFile(filePath, target);
+			stub.callCount.should.equal(1);
+			stub.getCall(0).args[0].should.equal(filePath);
+			stub.getCall(0).args[1].should.equal(target);
+		});
+		it('makeDirectory ', async () => {
+			const target = __dirname; //arbitrary path
+			sinon.stub(fs.promises, 'access').rejects({ code: 'ENOENT' }); // this simulates the target directory to not Exist.
+			const stub = sinon.stub(fs.promises, 'mkdir').resolves();
+			await platformService.makeDirectory(target);
+			stub.callCount.should.equal(1);
+			stub.getCall(0).args[0].should.equal(target);
+		});
+	});
+	it('showErrorMessage', () => {
+		const error = __dirname; //arbitrary known string
+		const stub = sinon.stub(vscode.window, 'showErrorMessage').resolves(); //resolves with a known string
+		platformService.showErrorMessage(error);
+		stub.callCount.should.equal(1);
+		stub.getCall(0).args[0].should.equal(error);
+	});
+	describe('isNotebookNameUsed', () => {
+		[true, false].forEach((isUsed => {
+			it(`return value: ${isUsed}`, () => {
+				const title = __filename; //arbitrary known string
+				if (isUsed) {
+					sinon.stub(azdata.nb, 'notebookDocuments').get(() => [{ isUntitled: true, fileName: title }]);
+					sinon.stub(vscode.workspace, 'textDocuments').get(() => [{ isUntitled: true, fileName: title }]);
+				} else {
+					sinon.stub(azdata.nb, 'notebookDocuments').get(() => [{ isUntitled: true, fileName: '' }]);
+					sinon.stub(vscode.workspace, 'textDocuments').get(() => [{ isUntitled: true, fileName: '' }]);
+				}
+				const result = platformService.isNotebookNameUsed(title);
+				result.should.equal(isUsed);
+			});
+		}));
+	});
+	describe('runCommand', () => {
+		[
+			//[commandSucceeds, ignoreError]
```

Comment:
For this kind of thing I prefer to use objects so you can use named fields : 

```
[
{ commandSucceeds: true, ignoreError: undefined},
...
]
```

Reply:
Good suggesion. Changed.